editor/scene_tree_dock.cpp

```diff
@@ -2064,9 +2064,21 @@ void SceneTreeDock::replace_node(Node *p_node, Node *p_by_node, bool p_keep_prop
 			if (!(E->get().usage & PROPERTY_USAGE_STORAGE)) {
 				continue;
 			}
+
 			if (E->get().name == "__meta__") {
+				if (ClassDB::is_parent_class(newnode->get_class(), "CanvasItem")) {
```

Comment:
Casting might be faster (and won't rely on strings): `Object::cast_to<CanvasItem>(newnode)`.

But maybe there would be a case for simply always keeping the `__meta__` instead of having to whitelist all the properties that are compatible? Or are there properties that would be problematic to keep?

Reply:
The cast makes sense. But the problem of always keeping the meta is that if you change that node again to a canvas one, the lock status are placed back, and it's kinda confusing as they were supposed to be gone when it was changed to a non-canvas node.